assessments/projects/opa/self-assessment.md

```diff
@@ -0,0 +1,563 @@
+policies in such a way that there are not problematic interactions
+
+The following issues raised during the assessment have been addressed:
+
+* OPA decision logs may contain sensitive information like usernames, passwords
+provided in the input to the policy. OPA now allows the user to mask sensitive
+fields in the input from showing up in the decision logs by allowing the user
+to define a policy that specifies the fields to mask
+
+* To make it easier for users to get familiar with Rego and also to make the
+policy authoring experience convenient and less error prone, we
+created [The Rego Playground](https://play.openpolicyagent.org/) and more
+recently we’ve updated the [OPA documentation](https://www.openpolicyagent.org/docs/latest/policy-language/)
+to make the policy examples interactive !
+
+### Compensating Mechanisms
+
+* The OPA daemon can be configured to authenticate and authorize requests. This
+could prove beneficial especially in scenarios where the deployment environment
+is untrusted. More details on deploying OPA securely can be found [here](https://www.openpolicyagent.org/docs/latest/security/). Additionally OAuth2.0 client credentials can be used to authenticate
+OPA’s REST APIs. See [this](https://github.com/open-policy-agent/opa/issues/1205)
+issue.
+
+* Since OPA does not contain any policies by default when started, it’s
+recommended that the application fail-open in case of any non-200 response
+from OPA.
+
+* To prevent an attacker from loading corrupt policies into OPA, users are
+recommended to sign their policies using a mechanism that best suits their
+use-case. Related issue [here](https://github.com/open-policy-agent/opa/issues/1757).
+
+* Since large bundles of policy and data have the potential to consume all of
+the host’s memory, OPA puts a restriction on the size of bundles that can
+be downloaded.
+
+## Secure development practices
+
+OPA has a well documented development workflow which can be
+found [here](https://github.com/open-policy-agent/opa/blob/master/docs/devel/DEVELOPMENT.md).
+
+Some highlights:
+
+* All source code is checked into GitHub
+
+* Every pull request kicks off a build which performs unit tests, benchmark
+tests and language best practices enforcement
+
+* Each pull request requires an approval from a core member before merging
+
+    * The reviewer makes sure all checks and tests are passing
+
+    * The reviewer pays close attention to any new exported methods or
+	interface definitions
+
+    * The reviewer sees that the code follows standard coding practices such as
+	adding comments, grouping common logic in functions
+
+    * Based on the changes made, reviewer sees that the docs are updated
+	accordingly
+
+* Pushes to master are forbidden by convention
+
+* OPA is at the passing criteria in the [Core Infrastructure Initiative (CII) badging process](https://bestpractices.coreinfrastructure.org/en/projects/1768)
```

Comment:
While I concur that OPA's entry in the CII shows passing, I did a review of the detailed requirements per the [worksheet here](https://github.com/cncf/sig-security/issues/153#issuecomment-487657305) and there were some items I felt might need clarification or supporting details to accurately assert that they are met.  That said, I don't think any of them would block this assessment as currently defined.  

I would suggest changing the wording to "OPA *asserts* that it meets the criteria..."

Reply:
Updated the wording.